drivers/cc1200/cc1200-netdev.c

```diff
@@ -0,0 +1,289 @@
+            cc1200->pan_id = *(uint16_t *) value;
+            break;
+        case NETOPT_ADDRESS_LONG:
+            if (value_len > sizeof(uint64_t)) {
+                res = -EOVERFLOW;
+            }
+            else {
+                res = cc1200_set_address_long(cc1200, *((uint64_t *)value));
+            }
+            break;
+#ifdef MODULE_GNRC_NETIF
+        case NETOPT_PROTO:
+            if (value_len != sizeof(gnrc_nettype_t)) {
+                return -EINVAL;
+            }
+            else {
+                cc1200->proto = (gnrc_nettype_t) value;
+                return sizeof(gnrc_nettype_t);
+            }
+            break;
+#endif
+        default:
+            break;
+    }
+
+    if (res == -ENOTSUP) {
+        res = netdev_ieee802154_set((netdev_ieee802154_t *)dev, opt,
+                                    value, value_len);
+    }
+    return res;
+}
+
+static void _netdev_cc1200_isr(void *arg)
+{
+    DEBUG("%s:%u\n", __func__, __LINE__);
+    netdev_t *netdev = (netdev_t *) arg;
+    unsigned state = irq_disable();
+    netdev->event_callback(netdev, NETDEV_EVENT_ISR);
+    irq_restore(state);
+    DEBUG("%s:%u\n", __func__, __LINE__);
+}
+
+static void _netdev_cc1200_rx_callback(void *arg)
+{
+    DEBUG("%s:%u\n", __func__, __LINE__);
+    netdev_t *netdev = (netdev_t *) arg;
+    cc1200_t *cc1200 = &((netdev_cc1200_t *) arg)->cc1200;
+    gpio_irq_disable(cc1200->params.int_pin);
+    netdev->event_callback(netdev, NETDEV_EVENT_RX_COMPLETE);
+}
+
+static void _isr(netdev_t *dev)
+{
+    DEBUG("%s:%u\n", __func__, __LINE__);
+    cc1200_t *cc1200 = &((netdev_cc1200_t *) dev)->cc1200;
+    unsigned state = irq_disable();
+    cc1200_isr_handler(cc1200, _netdev_cc1200_rx_callback, (void *)dev);
+    irq_restore(state);
+}
+
+static int _init(netdev_t *dev)
+{
+    DEBUG("%s:%u\n", __func__, __LINE__);
+    cc1200_t *cc1200 = &((netdev_cc1200_t *) dev)->cc1200;
+
+    gpio_init_int(cc1200->params.int_pin, GPIO_IN_PD, GPIO_BOTH,
+                  &_netdev_cc1200_isr, (void *)dev);
+    //gpio_init(cc1200->params.gdo0, GPIO_IN_PD);
+
+    gpio_set(cc1200->params.int_pin);
+    gpio_irq_disable(cc1200->params.int_pin);
+
+    /* Switch to RX mode */
+    cc1200_rd_set_mode(cc1200, RADIO_MODE_OFF);
+    cc1200_rd_set_mode(cc1200, RADIO_MODE_ON);
+    DEBUG("%s:%u\n", __func__, __LINE__);
+
+    return 0;
+}
+
+const netdev_driver_t netdev_cc1200_driver = {
+    .get = _get,
+    .set = _set,
+    .send = _send,
+    .recv = _recv,
+    .isr = _isr,
+    .init = _init
+};
+
+int netdev_cc1200_setup(netdev_cc1200_t *netdev_cc1200, const cc1200_params_t *params)
+{
+    DEBUG("netdev_cc1200_setup()\n");
+    //netdev_cc1200->netdev.netdev.driver = &netdev_cc1200_driver;
+    netdev_t *netdev = (netdev_t *)netdev_cc1200;
+
+    netdev->driver = &netdev_cc1200_driver;
+# ifdef MODULE_GNRC_SIXLOWPAN
+    netdev_cc1200->cc1200.proto = GNRC_NETTYPE_SIXLOWPAN;
+    netdev_cc1200->netdev.proto = GNRC_NETTYPE_SIXLOWPAN;
```

Comment:
I'm a bit confused here. I would have assumed setting `netdev_cc1200->netdev.proto` is enough. Is there duplication in `netdev_cc1200_t`?

Reply:
That is a good question. The duplication initially was constructed in case someone is trying to use the driver without the IEEE802.15.4 stack, as it was done with the CC110x driver. But I don't know how necessary that is.